src/v/cluster/types.h

```diff
@@ -1023,21 +1023,33 @@ struct join_reply
 ///   in future.
 struct join_node_request
   : serde::
-      envelope<join_node_request, serde::version<0>, serde::compat_version<0>> {
+      envelope<join_node_request, serde::version<1>, serde::compat_version<0>> {
     join_node_request() noexcept = default;
 
     explicit join_node_request(
-      cluster_version lv, std::vector<uint8_t> nuuid, model::broker b)
-      : logical_version(lv)
+      cluster_version lv,
+      cluster_version ev,
+      std::vector<uint8_t> nuuid,
+      model::broker b)
+      : latest_logical_version(lv)
+      , earliest_logical_version(ev)
       , node_uuid(nuuid)
       , node(std::move(b)) {}
 
-    explicit join_node_request(cluster_version lv, model::broker b)
-      : logical_version(lv)
+    explicit join_node_request(
+      cluster_version lv, cluster_version ev, model::broker b)
+      : latest_logical_version(lv)
+      , earliest_logical_version(ev)
       , node(std::move(b)) {}
 
     static constexpr int8_t current_version = 1;
-    cluster_version logical_version;
+
+    // The highest version that the joining node supports
+    cluster_version latest_logical_version{cluster::invalid_version};
+
+    // The lowest version that the joining node supports, it will already
+    // have its feature table initialized to this version.
+    cluster_version earliest_logical_version{cluster::invalid_version};
```

Comment:
IIRC the typical guidance for adding fields to envelopes is that if we don't want to bump the compat_version, this would need to be added to the end of the struct.